src/Jackett.Common/Indexers/Anidex.cs

```diff
@@ -219,55 +213,14 @@ private IEnumerable<ReleaseInfo> ParseResult(string response)
 
         private async Task ConfigureDDoSGuardCookie()
         {
-            const string pathAndQueryBase64Encoded = "Lw=="; // "/"
-            const string baseUriBase64Encoded = "aHR0cHM6Ly9hbmlkZXguaW5mbw=="; // "http://anidex.info"
-            const string ddosPostUrl = "https://ddgu.ddos-guard.net/ddgu/";
-
-            try
-            {
-                // Check if the cookie already exists, if so exit without doing anything
-                if (IsCookiePresent("__ddgu") && IsCookiePresent("__ddg1"))
-                {
-                    logger.Debug("DDOS Guard cookies are already present. Skipping bypass.");
-                    return;
-                }
-
-                // Make a request to DDoS Guard to get the redirect URL
-                var ddosPostData = new List<KeyValuePair<string, string>>
-                {
-                    { "u", pathAndQueryBase64Encoded },
-                    { "h", baseUriBase64Encoded },
-                    { "p", string.Empty }
-                };
-
-                var result = await PostDataWithCookiesAndRetry(ddosPostUrl, ddosPostData);
-
-                if (!result.IsRedirect)
-                    // Success returns a redirect. For anything else, assume a failure.
-                    throw new IOException($"Unexpected result from DDOS Guard while attempting to bypass: {result.Content}");
```

Comment:
I prefer the Exception type being differentiated like this, it allows us to catch specific errors instead of a `catch (Exception)` that has to figure if the exception we caught is the one we wanted to catch. Not sure if IOException is the best option though. The MS guideline on choosing an exception type can be found here: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/using-standard-exception-types

Reply:
The recommended way in all languages is to define your own exceptions extending Exception. That allows to catch exceptions in a more fine-grain.
In this case I don't think this is useful because I don't want to catch different exceptions because I don't want to do different things depending of the Exception type. Or all the code works or I want to fail. Throwing and Exception here is just a shortcut to stop the search execution and to log the reason.
I can change `IOException` => `Exception` or define a generic `DDOSGuardException` in this class. I prefer the first.

Reply:
@cadatoiva what should I do?

Reply:
Sorry, been mostly AFK today, with homeschooling my daughter. When StyleCop gets turned on in full, throwing `Exception` and catching `Exception` without rethrowing will result in build errors, as a Microsoft C# best practice (not my rule, see the link). There are a large number of built in exception types we (as C# devs) allowed (and encouraged) to use, and it looks like `System.Net.WebException` is probably our best bet. https://docs.microsoft.com/en-us/dotnet/api/system.net.webexception?view=netframework-4.8

Reply:
Changed.